templates/zabbix_agentd.conf.erb

```diff
@@ -5,38 +5,42 @@
 
 ############ GENERAL PARAMETERS #################
 
+<% if @pidfile %>
 ### Option: PidFile
 #	Name of PID file.
 #
-PidFile=<%= @pidfile %>
+PidFile=<%= @pidfile %><% end %>
 
 <% if @zabbix_version.to_f >= 3.0 -%>
+<% if @logtype %>
 ### Option: LogType
 #	Specifies where log messages are written to:
 #		system  - syslog
 #		file    - file specified with LogFile parameter
 #		console - standard output
 #
-<% if @logtype %>LogType=<%= @logtype %><% end %>
+LogType=<%= @logtype %><% end %>
 
+<% if @logfile %>
 ### Option: LogFile
 #	Log file name for LogType 'file' parameter.
 #
-LogFile=<%= @logfile %>
+LogFile=<%= @logfile %><% end %>
 
 <% else %>
+<% if @logfile %>
 ### Option: LogFile
 #	Name of log file.
 #	If not set, syslog is used.
 #
-LogFile=<%= @logfile %>
+LogFile=<%= @logfile %><% end %>
 <% end %>
 
 ### Option: LogFileSize
```

Comment:
Would it not make sense to wrap the if around the comment block so it doesn't appear if you don't set it?